snuba/cli/query_comparer.py

```diff
@@ -0,0 +1,292 @@
+    result_rows: MismatchedValues
+    result_bytes: MismatchedValues
+    read_rows: MismatchedValues
+    read_bytes: MismatchedValues
+
+    def performance_mismatches(self) -> PerformanceMismatchResult:
+        return PerformanceMismatchResult(
+            self.query_id,
+            self.query_duration_ms.base_value,
+            self.query_duration_ms.new_value,
+            self.read_rows.base_value,
+            self.read_rows.new_value,
+            self.read_bytes.base_value,
+            self.read_bytes.new_value,
+            self.query_duration_ms.delta,
+            self.read_rows.delta,
+            self.read_bytes.delta,
+        )
+
+    def data_mismatches(self) -> DataMismatchResult:
+        return DataMismatchResult(
+            self.query_id,
+            self.result_rows.base_value,
+            self.result_rows.new_value,
+            self.result_bytes.base_value,
+            self.result_bytes.new_value,
+            self.result_rows.delta,
+            self.result_bytes.delta,
+        )
+
+
+MEASUREMENTS = [
+    "query_duration_ms",
+    "result_rows",
+    "result_bytes",
+    "read_rows",
+    "read_bytes",
+]
+
+
+def write_querylog_comparison_results_to_csv(
+    results: Union[Sequence[DataMismatchResult], Sequence[PerformanceMismatchResult]],
+    filename: str,
+    header_row: Sequence[str],
+) -> None:
+    with open(filename, mode="w") as file:
+        writer = csv.writer(file)
+        writer.writerow(header_row)
+        for row in results:
+            writer.writerow(row)
+
+
+@click.command()
+@click.option(
+    "--base-file",
+    help="Clickhouse queries results from base version.",
+    required=True,
+)
+@click.option(
+    "--upgrade-file",
+    help="Clickhouse queries results from upgrade version.",
+    required=True,
+)
+@click.option(
+    "--table",
+    help="Clickhouse queries results from upgrade version.",
+    required=True,
+)
+@click.option("--log-level", help="Logging level to use.")
+def query_comparer(
+    *,
+    base_file: str,
+    upgrade_file: str,
+    table: str,
+    log_level: Optional[str] = None,
+) -> None:
+    """
+    compare results
+    """
+    setup_logging(log_level)
+    setup_sentry()
+
+    base = open(base_file)
+    upgrade = open(upgrade_file)
+
+    base_reader = csv.reader(base)
+    upgrade_reader = csv.reader(upgrade)
+
+    def query_measurement(row: Any) -> QueryMeasurement:
+        return QueryMeasurement(
+            query_id=row[0],
+            query_duration_ms=int(row[1]),
+            result_rows=int(row[2]),
+            result_bytes=int(row[3]),
+            read_rows=int(row[4]),
+            read_bytes=int(row[5]),
+        )
+
+    mismatches = []
+    total_rows = 0
+    for v1_row, v2_row in itertools.zip_longest(base_reader, upgrade_reader):
+        if v1_row[0] == "query_id":
+            # csv header row
+            continue
+
+        assert v1_row[0] == v2_row[0], "Invalid query_ids: ids must match"
```

Comment:
not right now, but that should be an easy change to the clickhouse query to the querylog to add ORDER BY query_id in

Reply:
Ok. Since we are iterating through one line through both of the files, I was thinking whether there could be a case of mismatches in the 2 files for some reason. Say 1 file missing 1 row for any reason. Not sure whether that could happen or not. But we can fix it later if we find this is happening.